### PR TITLE
Enable remote logging

### LINKED ISSUE #21: Extending support for remote MLFlow server
I would like to open a conversion about adding support for remote MLFlow server.

User story: As a corporate employee, I'd like to be able to send MLFlow tracking information to our company server to have experiments across all teams in one place.

Feature needed: Adding basic auth or bearer token to the header of the HTTP requests.

1, Would you be interested in adding it? I'm happy to open a PR.
2, What would be the suggested implementation?

Re 2)
I would propose to:
-  extend the MLFlow struct with an auth token in [types.jl](https://github.com/JuliaAI/MLFlowClient.jl/blob/main/src/types.jl) - add `headers()` method (similar to `uri()` method) that would set up the right headers for `mlfget()` and `mlfpost()` methods in [utils.jl](https://github.com/JuliaAI/MLFlowClient.jl/blob/main/src/utils.jl)

What do you think?

### PR DESCRIPTION
**Pull request by svilupp**
TL;DR This is a proposal to enable remote logging (eg, Databricks or internal MLFlow servers)

Closes #21 

Implementation: 
- It extends the basic constructor with a keyword field `headers`, where user can provide authentication details for their remote server (eg, basic auth, bearer)
- Given its sensitive nature, this field `headers` has been hidden from the pretty-print (ShowCase)

### CODE REVIEW

**Comment by svilupp** [comment]
The overall test suite keeps failing but that should be independent of my changes. I'll investigate it tomorrow.

**Comment by svilupp** [comment]
This PR is ready for review. It would be awesome if you could take a look @deyandyankov .

I've added docstrings and tests for the new method.

For visibility, the relevant tests pass but the overall test suite fails, because my computer blocks localhost (not sure how common/uncommon that is), but everything passes if I changed it to standard 127.0.0.1 (see [tests fix branch](https://github.com/svilupp/MLFlowClient.jl/tree/fix-test-suite))

**Review by deyandyankov — APPROVED** [review]
this looks amazing, thanks for submitting the PR!

**Comment by deyandyankov** [comment]
this is now merged, I've bumped the version to 0.4.0 and have submitted the new version to be registered in General. Thank you for your contribution!

**Comment by deyandyankov** [comment]
Thanks for the PR! I'll review this evening.
